ingest/scripts/add-host-categories.py

```diff
@@ -0,0 +1,70 @@
+#! /usr/bin/env python3
+"""
+From stdin, generates host names using info from the NCBI taxonomy output of the NDJSON record, with output to 'host'
+
+Outputs the modified record to stdout.
+"""
+
+import argparse
+import json
+from sys import stdin, stdout
+
+def parse_args():
+    parser = argparse.ArgumentParser(
+        description="Generate host names and output to 'host'.")
+    parser.add_argument("--latin-field", default='host_latin_name',
+        help="Field from the records to use as the host latin name.")
+    parser.add_argument("--family-field", default='host_family',
+        help="Field from the records to use as the host Family name.")
+    parser.add_argument("--genus-field", default='host_genus',
+        help="Field from the records to use as the host genus name.")
+    parser.add_argument("--group-field", default='host_group',
+        help="Field from the records to use as the host group.")
+    return parser.parse_args()
+
+def _set_host_name_transformed(record, args):
+    latin_replacements = {
+        "Canis lupus familiaris": "Domestic Dog",
+        "Homo sapiens": "Human",
+        "Bos taurus": "Cattle",
+        "Didelphis albiventris": "Other Mammal",
+        "Elephas maximus": "Other Mammal",
+        "Dasypus novemcinctus": "Other Mammal"}
+    family_replacements = {"Mephitidae": "Skunk"}
+    group_replacements = {
+        "odd-toed ungulates": "Other Ungulate",
+        "even-toed ungulates & whales": "Other Ungulate",
+        "carnivores": "Other Carnivore",
+        "bats": "Bat",
+        "birds": "Bird",
+        "primates": "Other Mammal",
+        "rodents": "Other Mammal",
+        "mammals": "Other Mammal"
+    }
```

Comment:
non-blocking, but this would end up being more generic (and hence easier to ultimately move into augur or re-use in other repos) if these were provided via config files passed as CLI args.

Reply:
For sure! I wasn't sure how to make the config file generalized, but it [just occurred to me](https://github.com/nextstrain/rabies/pull/8#discussion_r1723730907) that we can borrow the format of the hierarchical geolocation rules.

Reply:
Let's discuss the generic version in https://github.com/nextstrain/augur/issues/1586